doc/whatsnew/fragments/10482.false_negative

```diff
@@ -0,0 +1,3 @@
+Fix false-negative for ``used-before-assignment`` with ``from __future__ import annotations`` in function definitions.
```

Comment:
```suggestion
Fix false-negative for ``used-before-assignment`` with ``from __future__ import annotations`` in function definitions for python 3.14
```
Unless I'm mistaken?

Reply:
No, this is actually a false-negative for all versions. The 3.14 PR just happened to uncover it as 3.14 enables deferred evaluations of annotations for everything.

_That's also why I added the backport label just in case we do release another bug fix release before 4.0._